### PR TITLE
Bump package versions to latest

### LINKED ISSUE #1044: Microsoft.DotNet.TestHost.1.0.3-prerelease only targets 32-bit
Running tests on Window 7 64-bit fails due to `CoreRun.exe` being 64-bit and the TestHost dlls being 32-bit.

When setting the 32-bit flag on `xunit.console.netcore.exe` with `corflags.exe` and invoking it directly (iow without `corerun.exe`), the tests work again.

    > dumpbin /headers CoreRun.exe | find "machine"
            8664 machine (x64)

    > dumpbin /headers api-ms-win-core-console-l1-1-0.dll | find "machine"
             14C machine (x86)

### PR DESCRIPTION
**Pull request by nguerrera**
NOTE: Will not pass CI until buildtools version 15 are published.

Fixes #1044 and unblocks #1076 

cc @weshaggard @tarekgh @jhendrixMSFT

### CODE REVIEW

**Comment by krwq** [comment]
Looks good :+1:
Only this: System.InvalidOperationException: Unable to find version '1.0.25-prerelease-00015' of package 'Microsoft.DotNet.BuildTools'.

 It would be nice if the version of the package (at least unstable) contained a date so we would know when was the package produced without having to go to the nuget/myget.org

File: src/System.Collections.NonGeneric/src/packages.config [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  <package id="System.Diagnostics.Contracts" version="4.0.0-beta-22703" targetFramework="portable-net45+win" />
+  <package id="System.Diagnostics.Debug" version="4.0.10-beta-22703" targetFramework="portable-net45+win" />
+  <package id="System.Diagnostics.Tools" version="4.0.0-beta-22703" targetFramework="portable-net45+win" />
+  <package id="System.Globalization" version="4.0.10-beta-22703" />
+  <package id="System.Reflection" version="4.0.10-beta-22703" targetFramework="portable-net45+win" />
+  <package id="System.Reflection.Primitives" version="4.0.0-beta-22703" targetFramework="portable-net45+win" />
+  <package id="System.Resources.ResourceManager" version="4.0.0-beta-22703" />
+  <package id="System.Runtime" version="4.0.20-beta-22703" />
+  <package id="System.Runtime.Extensions" version="4.0.10-beta-22703" />
+  <package id="System.Threading" version="4.0.10-beta-22703" />
```
Comment (krwq) [inline src/System.Collections.NonGeneric/src/packages.config]: Can we reference only newest version or both 4.0.0-XXX and 4.0.10-XXX or will it just work if the contract is satisified?

**Comment by tarekgh** [comment]
LGTM too

File: src/System.Collections.NonGeneric/src/packages.config (hunk 1 above)
Comment (nguerrera) [inline src/System.Collections.NonGeneric/src/packages.config]: Intentional. Only newest version for now. We will have a story for referencing 4.0.0 before long.

**Comment by nguerrera** [comment]
Rebased to address merge conflict.

**Pull request merged**